packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -242,9 +242,13 @@ function setSelected(el: HTMLSelectElement, value: any, number: boolean) {
         const optionType = typeof optionValue
         // fast path for string / number values
         if (optionType === 'string' || optionType === 'number') {
-          option.selected = value.includes(
-            number ? looseToNumber(optionValue) : optionValue,
-          )
+          option.selected = !!value.find(v => {
+            let value = optionValue
+            if (number) {
+              value = looseToNumber(optionValue)
+            }
+            return looseEqual(value, v)
```

Comment:
This doesn't seem right, and if you want to call `looseEqual`, the fast path here doesn't make sense.

see https://github.com/vuejs/core/commit/2ffb956efe692da059f4895669084c5278871351
![image](https://github.com/vuejs/core/assets/3705199/24d194e0-7b78-43da-adf7-53143c825b3d)

Reply:
You're right, awesome

Reply:
```ts
// fast path for number values
// there are probably lots of use case like #10574, so for string values should 
// compare using `looseEqual` to avoid regression
if (optionType === 'number') {
  option.selected = value.includes(
    number ? looseToNumber(optionValue) : optionValue,
  )
} else {
  option.selected = looseIndexOf(value, optionValue) > -1
}
```

Reply:
Maybe both the string type and number type should retain the fast path. I added type judgment and I don’t know whether it will have an impact on the fast channel.

Reply:
- fast path means without using `looseEqual`
- Your code does the same thing as `looseIndexOf`

Reply:
I'm aware of that, I added a new commit to handle the same type directly, and only use looseEqual when it's different.

Reply:
how about this?
```js
option.selected = value.some(v => String(v)===String(optionValue))
```

Reply:
That makes sense, my brother. Why didn't I think of that?